### PR TITLE
Preserve attributes order for element node

### PR DESCRIPTION
**Pull request by dcyriller**
## Description

Closes https://github.com/prettier/prettier/issues/10203

## Checklist

- [x] I’ve added tests to confirm my change works.
- [ ] (If changing the API or CLI) I’ve documented the changes I’ve made (in the `docs/` directory).
- [x] (If the change is user-facing) I’ve added my changes to `changelog_unreleased/*/XXXX.md` file following `changelog_unreleased/TEMPLATE.md`.
- [x] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/main/CONTRIBUTING.md).

**✨[Try the playground for this PR](https://prettier.io/playground-redirect)✨**

### CODE REVIEW

File: src/language-handlebars/printer-glimmer.js [hunk 1]
```diff
@@ -418,24 +418,50 @@ function print(path, options, print) {
 
 /* ElementNode print helpers */
 
+function sortByLoc(a, b) {
+  if (a.loc.start.line < b.loc.start.line) {
```
Comment (fisker) [inline src/language-handlebars/printer-glimmer.js:422]: `return (a.loc.start.line - b.loc.start.line) || (a.loc.start.column - b.loc.start.column)`

File: src/language-handlebars/printer-glimmer.js [hunk 2]
```diff
@@ -418,24 +418,50 @@ function print(path, options, print) {
...
-      line,
-      property === "blockParams"
-        ? printBlockParams(node)
-        : join(line, path.map(print, property)),
-    ]);
+  const types = ["attributes", "modifiers", "comments"].filter((property) =>
+    isNonEmptyArray(node[property])
+  );
+  const attributes = types
+    .reduce((acc, type) => [...acc, ...node[type]], [])
```
Comment (fisker) [inline src/language-handlebars/printer-glimmer.js:450]:
```suggettion
    .flatMap((type) => node[type])
```

File: changelog_unreleased/handlebars/10958.md [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+  @arg1="hello"
+  {{on "clik" this.modify}}
+  @arg2="hello"
+  {{! this is a comment for arg 3}}
+  @arg3="hello"
+  @arg4="hello"
+  {{! this is a comment for arg 5}}
+  @arg5="hello"
+  ...arguments
+/>
```
Comment (fisker) [inline changelog_unreleased/handlebars/10958.md:17]: Can we make this example a little shorter?

**Review by fisker — APPROVED** [review]

**Comment by dcyriller** [comment]
CI failure seems unrelated

File: changelog_unreleased/handlebars/10958.md (hunk 3 above)
Comment (thorn0) [reply changelog_unreleased/handlebars/10958.md:17]: Oops. Sorry, missed this comment, but IMHO it's good as is.

Comment (dcyriller) [reply changelog_unreleased/handlebars/10958.md:17]: It is odd, I thought I applied the change. If you think it's worth correcting, I'll open a PR

Comment (thorn0) [reply changelog_unreleased/handlebars/10958.md:17]: It'll be a minor release without a blog post, so it's okay, don't bother

**Pull request merged**